core/frontend/src/main/scala/io/udash/bindings/inputs/Input.scala

```diff
@@ -0,0 +1,79 @@
+package io.udash.bindings.inputs
+
+import io.udash._
+import org.scalajs.dom.{Element, Event, KeyboardEvent}
+import org.scalajs.dom.html.{Input => JSInput}
+import scalatags.JsDom
+import scalatags.JsDom.all._
+
+import scala.concurrent.duration.{Duration, DurationInt}
+
+/** Abstraction for HTML input tags.*/
+private[bindings] abstract class Input(inputType: String) {
+  /**
+    * @param value Property to bind.
+    * @param debounce Property update timeout after input changes.
+    * @param xs Additional Modifiers, don't use modifiers on value, onchange and onkeyup attributes.
+    * @return HTML input with bound Property, applied modifiers and nested options.
+    */
+  def apply(value: Property[String], debounce: Duration = 20 millis)(xs: Modifier*): InputBinding[JSInput] =
+    new InputBinding[JSInput] {
+      private val element = input(
+        xs, tpe := inputType,
+        nestedInterceptor(new InputModifier(value, Some(debounce)))
+      ).render
+
+      override def render: JSInput = element
+    }
+
+  /**
+    * @param property Property to bind.
+    * @param debounce Property update timeout after input changes.
+    * @param xs Additional Modifiers, don't use modifiers on value, onchange and onkeyup attributes.
+    * @return HTML textarea with bound Property, applied modifiers and nested options.
+    */
+  @deprecated("Use `apply` returning `InputBinding` instead.", "0.7.0")
```

Comment:
You can probably just remove these methods since this object is `private[bindings]`

Reply:
This private abstract class is extended by public objects `TextInput`, `PasswordInput` and `NumberInput`.